app/models/ability.rb

```diff
@@ -0,0 +1,13 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    p 'abilit ---', user
+    return unless user.present?
+
```

Comment:
- [ ] According to the project requirement and to match the expectation result, I think it would be better if you update this code lines by checking if the user has the role of `admin` instead of checking if he is present. for example:
```ruby
class Ability
 include CanCan::Ability

     def initialize(user)

         user ||= User.new 

         if user.admin? 
           can :manage, :all
         else
           # The rest of your code here
           ...
         end
     end
 end
```